File: Back-Ends/TypeScript/src/Module/account/models/account.model.ts
import Sequelize from "sequelize";
import { Model } from "sequelize-typescript";
import { sequelize } from "../../../../Database/Sequelize";

export interface AccountModel extends Model {
  userId: string;
  balance: string;
  createdAt: Date;
  updatedAt: Date;
}

export const Account = sequelize.define<AccountModel>("accounts", {
  userId: {
    type: Sequelize.STRING,
    allowNull: false,
    unique: false,
    references: {
      model: "User",
      key: "id",
    },
    onUpdate: "CASCADE",
    onDelete: "CASCADE",
  },
  balance: {
    type: Sequelize.STRING,
    allowNull: false,
  },
  createdAt: Sequelize.DATE,
  updatedAt: Sequelize.DATE,
});

File: Back-Ends/Golang/api/models/account.model.go
package models

import (
	"time"
)

type Account struct {
	ID        uint32    `gorm:"primary_key;auto_increment" json:"id"`
	UserId    uint32    `sql:"type:int REFERENCES users(id)" gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
	Balance   string    `gorm:"size:100;not null;default:0" json:"balance"`
	CreatedAt time.Time `gorm:"default:CURRENT_TIMESTAMP" json:"createdAt"`
	UpdatedAt time.Time `gorm:"default:CURRENT_TIMESTAMP" json:"updatedAt"`
}
